packages/elements/src/clock/index.ts

```diff
@@ -479,46 +568,69 @@ export class Clock extends ResponsiveElement {
 
   /**
   * Handles UP key press
-  * @param event Event Object
   * @returns {void}
   */
-  private handleUpKey (event: KeyboardEvent): void {
-    this.shift(UP, this.getShiftAmountFromTarget(event.target));
+  private handleUpKey (): void {
+    this.shift(Direction.UP, this.getShiftAmountFromSegment(this.activeSegment));
   }
 
   /**
   * Handle DOWN key press
-  * @param event Event Object
   * @returns {void}
   */
-  private handleDownKey (event: KeyboardEvent): void {
-    this.shift(DOWN, this.getShiftAmountFromTarget(event.target));
+  private handleDownKey (): void {
+    this.shift(Direction.DOWN, this.getShiftAmountFromSegment(this.activeSegment));
+  }
+
+  /**
+  * Updates aria-value-text to have the same value as screen
+  * @returns {void}
+  */
+  private async updateAriaValue () {
+    const value = await this.tPromise('TIME', {
+      value: parse(this.displayValue),
+      amPm: this.amPm,
+      showSeconds: this.showSeconds
+    });
+    this.setAttribute('aria-valuenow', `${this.displayTime}`);
+    this.setAttribute('aria-valuetext', value);
   }
 
   /**
   * Template for increment and decrement button
   * if interactive mode is enabled.
+  * @param segment active segment
   * @returns {TemplateResult} template result
   */
-  private generateButtonsTemplate (): TemplateResult {
+  private generateButtonsTemplate (segment: Segment): TemplateResult {
+    const isActive = this.focused && segment === this.activeSegment;
```

Comment:
Shouldn't this be the same as line 628 https://github.com/Refinitiv/refinitiv-ui/pull/308/files#diff-9683b0f1fc8b119d1e15fc2156fa96112e9615d2d14a6aa657a6f0598c5faf48R628 ?

Reply:
Would it be better if it becomes a common function ?

Reply:
Updated by remove the condition at `generateButtonTemplate`. Because `generateSegmentTemplate` is already checked